src/FastRandom.cpp

```diff
@@ -13,109 +13,49 @@
 #elif defined (_MSC_VER)
 	#define ATTRIBUTE_TLS static __declspec(thread)
 #else
-	#error "Unknown thread local storage qualifier"
+	#define ATTRIBUTE_TLS thread_local
 #endif
 
-static unsigned int GetRandomSeed()
-{
-	ATTRIBUTE_TLS bool SeedCounterInitialized = 0;
-	ATTRIBUTE_TLS unsigned int SeedCounter = 0;
-
-	if (!SeedCounterInitialized)
-	{
-		std::random_device rd;
-		std::uniform_int_distribution<unsigned int> dist;
-		SeedCounter = dist(rd);
-		SeedCounterInitialized = true;
-	}
-	return ++SeedCounter;
-}
-
-
-
-
-////////////////////////////////////////////////////////////////////////////////
-// cFastRandom:
-
-
-
-
-
-cFastRandom::cFastRandom(void) :
-	m_LinearRand(GetRandomSeed())
-{
-}
-
-
-
-
-
-int cFastRandom::NextInt(int a_Range)
-{
-	std::uniform_int_distribution<> distribution(0, a_Range - 1);
-	return distribution(m_LinearRand);
-}
-
-
-
-
-
-
-float cFastRandom::NextFloat(float a_Range)
-{
-	std::uniform_real_distribution<float> distribution(0, a_Range);
-	return distribution(m_LinearRand);
-}
+// This list allows deletion of elements as if they had static storage duration
+static std::mutex CSDeleteList;
+static std::list<std::unique_ptr<MTRand>> DeleteList;
 
 
 
 
 
-
-int cFastRandom::GenerateRandomInteger(int a_Begin, int a_End)
+/** Some compilers don't support thread_local for non-POD types, this is purely a work around for that restriction.
+There should be minimal overhead for the non-initializing case and all thread's instances are deleted properly. */
```

Comment:
This comment is not documentation, so it should not use the doxy-comment syntax. Also it should be inside the function, where the pointers are actually declared.